### PR TITLE
Config namespace fix and changelog update

### PR DESCRIPTION
**Pull request by lengau**
- [x] Have you followed the guidelines for contributing?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `tox`?

-----

### CODE REVIEW

File: docs/reference/changelog.rst [hunk 1]
```diff
@@ -2,6 +2,17 @@
...
 *********
 
+4.2.3 (2024-Sep-18)
+-------------------
+
+Application
+===========
+
+- get_arg_or_config now correctly checks the config service if the passed
+  namespace contains the given value but it is None.
```
Comment (medubelko) [inline docs/reference/changelog.rst:12]:
```suggestion
- ``get_arg_or_config`` now correctly checks the config service if the passed
  namespace contains the given value and the value is ``None``.
```

**Review by medubelko — CHANGES_REQUESTED** [review]

**Review by tigarmo — APPROVED** [review]

File: docs/reference/changelog.rst [hunk 2]
```diff
@@ -2,6 +2,17 @@
 Changelog
 *********
 
+4.2.3 (2024-Sep-18)
+-------------------
+
+Application
+===========
+
+- get_arg_or_config now correctly checks the config service if the passed
```
Comment (tigarmo) [inline docs/reference/changelog.rst:11]: Incidentally, I like this idea of adding the changelog entry on the PR with the change. Maybe we should start doing that on the libraries?

**Review by mr-cal — APPROVED** [review]

File: docs/reference/changelog.rst (hunk 1 above)
Comment (lengau) [reply docs/reference/changelog.rst]: The latter part of that makes it more confusing IMO. What about "if the passed namespace has ``None`` as the value of the requested item."

Comment (medubelko) [reply docs/reference/changelog.rst]: Yup, that works too.

**Pull request merged**